spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/bigquant/Linear.scala

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.bigquant
+
+import com.intel.analytics.bigdl.bigquant.BigQuant
+import com.intel.analytics.bigdl.nn.ErrorInfo
+import com.intel.analytics.bigdl.nn.abstractnn.TensorModule
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.tensor.{FloatType, QuantTensor, Tensor}
+import com.intel.analytics.bigdl.utils.serializer.{DataConverter, ModuleData, ModuleSerializable}
+import com.intel.analytics.bigdl.utils.{T, Table}
+import java.io.{IOException, ObjectInputStream, ObjectOutputStream}
+import scala.reflect.ClassTag
+import serialization.Bigdl.{AttrValue, BigDLModule, BigDLTensor, DataType}
+import scala.collection.JavaConverters._
+import scala.reflect.runtime.universe
+
+class Linear[T: ClassTag](
+  val inputSize: Int,
+  val outputSize: Int,
+  val withBias: Boolean = true
+)(implicit ev: TensorNumeric[T]) extends TensorModule[T] {
+
+  val data: QuantTensor[T] = QuantTensor[T]()
+  @transient var weight: QuantTensor[T] = _
+  val bias: Tensor[T] = Tensor[T](outputSize)
+
+  val weightSum = new Array[T](outputSize)
+  val min = new Array[T](outputSize)
+  val max = new Array[T](outputSize)
```

Comment:
We should make these (sum, min, max) a part of the QuantTensor, so that the applications don't need to handle them